File: backProject/src/main/java/org/example/backproject/entity/Posts.java
package org.example.backproject.entity;

import jakarta.persistence.*;
import lombok.*;
import org.example.backproject.repository.UsersRepository;
import org.springframework.beans.factory.annotation.Autowired;

import java.time.LocalDateTime;
import java.time.temporal.ChronoUnit;

import static jakarta.persistence.FetchType.LAZY;

@Entity
@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Table(name = "posts")
public class Posts {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = LAZY, optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    private Users user;



    @ManyToOne(fetch = LAZY)
    @JoinColumn(name = "restaurant_id")
    private Restaurants restaurant;

    private String restaurantName;
    private String latitude;
    private String longitude;

    private boolean visited = false;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    public void toggleVisited() {
        this.visited = !this.visited;
    }

    //    ❗❗❗❗❗개발용으로 추가한 생성자입니다.
    public Posts(Users user, Restaurants restaurant) {

        this.user = user;
        this.restaurant = restaurant;
        this.createdAt = LocalDateTime.now().truncatedTo(ChronoUnit.SECONDS); // 초 단위 절삭 적용
    }
}

File: frontProject/src/mocks/posts.ts
export interface Post {
  id: number;
  restaurant_id: number;
  user_id: number;
  restaurant_name: string;
  latitude: number;
  longitude: number;
  created_at: string;
}

export const posts: Post[] = [
  {
    id: 1,
    restaurant_id: 1,
    user_id: 1,
    restaurant_name: "맛있는 돈까스",
    latitude: 37.5665,
    longitude: 126.9780,
    created_at: "2024-03-15T14:30:00Z"
  },
  {
    id: 2,
    restaurant_id: 2,
    user_id: 2,
    restaurant_name: "신선한 초밥",
    latitude: 37.5666,
    longitude: 126.9781,
    created_at: "2024-03-14T18:45:00Z"
  }
]; 